plugins/cody-chat/src/com/sourcegraph/cody/chat/agent/CodyAgentClientImpl.java

```diff
@@ -28,8 +48,49 @@ public CompletableFuture<ProtocolTextDocument> textDocument_openUntitledDocument
 
   @Override
   public CompletableFuture<Boolean> textDocument_show(TextDocument_ShowParams params) {
-
-    return null;
+    CompletableFuture<Boolean> result = new CompletableFuture<>();
+    Display.getDefault()
+        .asyncExec(
+            () -> {
+              IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
+              if (window != null) {
+                IWorkbenchPage page = window.getActivePage();
+                if (page != null) {
+                  try {
+                    // Construct a URI from the params.uri string
+                    IFileStore fileStore = EFS.getStore(new URI(params.uri));
+                    var editor = IDE.openEditorOnFileStore(page, fileStore);
+                    if (editor instanceof ITextEditor
+                        && params.options != null
+                        && params.options.selection != null) {
+                      ITextEditor textEditor = (ITextEditor) editor;
+                      IDocument document =
+                          textEditor.getDocumentProvider().getDocument(textEditor.getEditorInput());
+
+                      TextSelection selection =
+                          new TextSelection(
+                              document,
+                              getOffset(document, params.options.selection.start),
+                              getOffset(document, params.options.selection.end)
+                                  - getOffset(document, params.options.selection.start));
+
+                      textEditor.selectAndReveal(selection.getOffset(), selection.getLength());
+                    }
+                    result.complete(true);
+                  } catch (Exception e) {
+                    // Handle other potential exceptions
+                    log.error("Error opening file", e);
+                    result.complete(false);
+                  }
+                }
+              }
+            });
+    return result;
+  }
```

Comment:
@Kordyjan this is my first foray into (admittedly trivial) actual Eclipse dev. So let me have it in terms of critique!